usr/lib/common/decr_mgr.c

```diff
@@ -546,6 +546,38 @@ CK_RV decr_mgr_init(STDLL_TokData_t *tokdata,
         }
         memset(ctx->context, 0x0, sizeof(AES_CONTEXT));
         break;
+    case CKM_AES_XTS:
+        // XXX Copied from DES3, should be verified - KEY
+        if (mech->ulParameterLen != AES_INIT_VECTOR_SIZE ||
+                    mech->pParameter == NULL) {
+                    TRACE_ERROR("%s\n", ock_err(ERR_MECHANISM_PARAM_INVALID));
+                    rc = CKR_MECHANISM_PARAM_INVALID;
+                    goto done;
+                }
+            // is the key type correct?
+            //
+            rc = template_attribute_get_ulong(key_obj->template, CKA_KEY_TYPE,
+                                              &keytype);
+            if (rc != CKR_OK) {
+                TRACE_ERROR("Could not find CKA_KEY_TYPE for the key.\n");
+                goto done;
+            }
+
+            if (keytype != CKK_AES_XTS) {
+                TRACE_ERROR("%s\n", ock_err(ERR_KEY_TYPE_INCONSISTENT));
+                rc = CKR_KEY_TYPE_INCONSISTENT;
+                goto done;
+            }
+
+            ctx->context_len = sizeof(AES_CONTEXT);
+            ctx->context = (CK_BYTE *) malloc(sizeof(AES_CONTEXT));
+            if (!ctx->context) {
```

Comment:
You don't seem to be using the `ctx->context` (aka AES_CONTEXT) anywhere in your added code.
For single part support you don't need that at all. Multi part support is completely missing, and this will probably need a different structure then.
So please remove the allocation of the context if you are not using it.

Reply:
ah never mind, you did implement multi part support....